app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/CreateDBTablePageSolution.java

```diff
@@ -0,0 +1,436 @@
+
+    private final String TEMPLATE_TABLE_NAME = "templateTableName";
+
+    private final String TEMPLATE_APPLICATION_FILE = "template application file";
+
+    private final String DELETE_FIELD = "deleteThisFieldFromActionsAndLayout";
+
+    private final String SELECT_QUERY = "selectQuery";
+
+    @Autowired
+    public CreateDBTablePageSolution(NewPageService newPageService,
+                                     DatasourceService datasourceService,
+                                     ApplicationService applicationService,
+                                     LayoutActionService layoutActionService,
+                                     ApplicationPageService applicationPageService) {
+        this.newPageService = newPageService;
+        this.datasourceService = datasourceService;
+        this.applicationService = applicationService;
+        this.layoutActionService = layoutActionService;
+        this.applicationPageService = applicationPageService;
+    }
+
+    // These fields contain the mapping fields between template DB table and DB table in current context
+    private final Set<String> WIDGET_FIELDS = Set.of("defaultText", "placeholderText", "text", "options", "defaultOptionValue");
+
+    public Mono<PageDTO> createPageFromDBTable(String pageId, Object requestBody) {
+
+        final Map<String, Object> tableObject = (HashMap<String, Object>) requestBody;
+        AtomicReference<String> savedPageId = new AtomicReference<>(pageId);
+        if (tableObject.get("tableName") == null || tableObject.get("datasourceName") == null) {
+            return Mono.error(new AppsmithException(AppsmithError.INVALID_PARAMETER, ", tableName and datasourceName must be present"));
+        } else if (tableObject.get(FieldName.APPLICATION_ID) == null) {
+            return Mono.error(new AppsmithException(AppsmithError.INVALID_PARAMETER, FieldName.APPLICATION_ID));
+        }
+
+        final String tableName =  tableObject.get("tableName").toString();
+        final String datasourceName =  tableObject.get("datasourceName").toString();
+        final String applicationId = tableObject.get(FieldName.APPLICATION_ID).toString();
+
+        //Mapped columns along with table name between template and concerned DB table
+        Map<String, String> mappedColumnsAndTableName = new HashMap<>();
+
+        Mono<NewPage> pageMono = getPage(applicationId, savedPageId.get(), tableName).cache();
+
+        Mono<Datasource> datasourceMono = pageMono
+            .flatMap(newPage -> {
+                savedPageId.set(newPage.getId());
+                return applicationService.findById(newPage.getApplicationId());
+            })
+            .flatMap(application ->
+                datasourceService
+                    .findAllByOrganizationId(application.getOrganizationId(), AclPermission.MANAGE_DATASOURCES)
+                    .filter(datasource -> datasource.getStructure() != null && !datasource.getStructure().getTables().isEmpty())
```

Comment:
Just so we don't get an NPE when `.getTables` is `null`.

```suggestion
                    .filter(datasource -> datasource.getStructure() != null && !CollectionUtils.isEmpty(datasource.getStructure().getTables()))
```